keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -0,0 +1,733 @@
++ COSI detects the existence of the BAR and follows the [grant access](#grant-bucket-access) steps.
++ COSI detects the existence of the BAR, BA and executes a validation process.
++ As a validation step, COSI ensures that the `BucketRequest.bucketInstanceName` matches the `BucketAccessRequest.bucketAccessName.bucketInstanceName`. In other words, the two related `Bucket` references match, meaning that the BR and BAR->BA point to the same `Bucket`.
+
+---
+
+# gRPC Definitions
+
+There is one service defined by COSI - `Provisioner`. This will need to be satisfied by the vendor-provisioner in order to be COSI-compatible
+
+```
+service Provisioner {
+    rpc ProvisionerGetInfo (ProvisionerGetInfoRequest) returns (ProvisionerGetInfoResponse) {}
+
+    rpc ProvisionerCreateBucket (ProvisionerCreateBucketRequest) returns (ProvisionerCreateBucketResponse) {}
+    rpc ProvisionerDeleteBucket (ProvisionerDeleteBucketRequest) returns (ProvisionerDeleteBucketResponse) {}
+
+    rpc ProvisionerGrantBucketAccess (ProvisionerGrantBucketAccessRequest) returns (ProvisionerGrantBucketAccessResponse);
+    rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
+}
+```
+
+## ProvisionerGetInfo
+
+This call is meant to retrieve the unique provisioner Identity. This identity will have to be set in `BucketRequest.Provisioner` field in order to invoke this specific provisioner.
+
+```
+message ProvisionerGetInfoRequest {
+    // Intentionally left blank
+}
+
+message ProvisionerGetInfoResponse {
+  // The name MUST follow domain name notation format
+  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
+  // include the plugin's host company name and the plugin name,
+  // to minimize the possibility of collisions. It MUST be 63
+  // characters or less, beginning and ending with an alphanumeric
+  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+  // alphanumerics between. This field is REQUIRED.
+  string name = 1;
+}
+```
+
+## ProvisonerCreateBucket
+
+This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner.
+
+```
+message ProvisionerCreateBucketRequest {    
+    // This field is REQUIRED
+    string bucket_name = 1;
+
+    map<string,string> bucket_context = 2;
+
+    enum AnonymousBucketAccessMode {
+	PRIVATE = 0;
+	PUBLIC_READ_ONLY = 1;
+	PUBLIC_WRITE_ONLY = 2;
+	PUBLIC_READ_WRITE = 3;
+    }
+    
+    AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
+}
+
+message ProvisionerCreateBucketResponse {
+    // Intentionally left blank
+}
+```
+
+## ProvisonerDeleteBucket
+
+This call is made to delete the bucket in the backend. If the bucket has already been deleted, then no error should be returned.
```

Comment:
Please add "ProvisonerDeleteBucket MUST be idempotent".

Reply:
Fixed